x/wasm-storage/types/filters_util.go

```diff
@@ -0,0 +1,59 @@
+package types
+
+import (
+	"encoding/base64"
+
+	"github.com/tidwall/gjson"
+)
+
+type dataAttributes struct {
+	freqMap map[string]int // frequency map from data to frequency
+	maxFreq int            // frequency of most frequent data in data list
+}
+
+// parseReveals parses a list of RevealBody objects using a given data
+// path and returns a data list. However, if more than 1/3 of the reveals
+// are corrupted (i.e. cannot be parsed), no data list is returned and
+// ErrCorruptReveals error is returned. When there is no error, it also
+// returns dataAttributes struct since some filters require this information.
+// Note that when an i-th reveal is corrupted, the i-th item in the data
+// list is left as an empty string.
+func parseReveals(reveals []RevealBody, dataPath string) ([]string, dataAttributes, error) {
+	if len(reveals) == 0 {
+		return nil, dataAttributes{}, ErrEmptyReveals
+	}
+
+	var maxFreq, corruptCount int
+	freq := make(map[string]int, len(reveals))
+	dataList := make([]string, len(reveals))
+	for i, r := range reveals {
+		if r.ExitCode != 0 {
+			corruptCount++
+			continue
+		}
+
+		revealBz, err := base64.StdEncoding.DecodeString(r.Reveal)
```

Comment:
What is revealBz?

Reply:
Cosmos uses `bz` for some reason for byte slices, but I realized it's just a Cosmos thing :D Will update this